nucypher/network/server.py

```diff
@@ -252,7 +252,13 @@ def reencrypt():
             # TODO (#567): bucket the node as suspicious.
             return Response(message, status=HTTPStatus.BAD_REQUEST)
 
-        # Enforce Reencryption Conditions
+        # Enforce Subscription Manager Payment
```

Comment:
Are we able to check this even earlier in the logic? For example, do it at L230?

The TODO at L211 can go away.

Reply:
Leaving payment and conditions evaluation here prevents the condition from being evaluated with bogus cryptographic material.